firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java

```diff
@@ -1131,6 +1135,87 @@ public void testAppDeleteLeadsToFirestoreTerminate() {
     assertTrue(instance.getClient().isTerminated());
   }
 
+  @Test
+  public void testDefaultNamedDbIsSame() {
+    FirebaseApp app = FirebaseApp.getInstance();
+    FirebaseFirestore db1 = FirebaseFirestore.getInstance();
+    FirebaseFirestore db2 = FirebaseFirestore.getInstance(app);
+    FirebaseFirestore db3 = FirebaseFirestore.getInstance(app, "(default)");
+    FirebaseFirestore db4 = FirebaseFirestore.getInstance("(default)");
+
+    assertSame(db1, db2);
+    assertSame(db1, db3);
+    assertSame(db1, db4);
+  }
+
+  @Test
+  public void testSameNamedDbIsSame() {
+    FirebaseApp app = FirebaseApp.getInstance();
+    FirebaseFirestore db1 = FirebaseFirestore.getInstance(app, "myDb");
+    FirebaseFirestore db2 = FirebaseFirestore.getInstance("myDb");
+
+    assertSame(db1, db2);
+  }
+
+  @Test
+  public void testDifferentDbNamesAreDifferent() {
+    FirebaseFirestore db1 = FirebaseFirestore.getInstance();
+    FirebaseFirestore db2 = FirebaseFirestore.getInstance("db1");
+    FirebaseFirestore db3 = FirebaseFirestore.getInstance("db2");
+
+    assertNotSame(db1, db2);
+    assertNotSame(db1, db3);
+    assertNotSame(db2, db3);
+  }
+
+  @Test
+  public void testNamedDbHaveDifferentPersistence()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    // TODO: Have backend with named databases created beforehand.
+    // Emulator doesn't care if database was created beforehand.
+    if (!isRunningAgainstEmulator()) {
```

Comment:
Use `org.junit.Assume.assumeTrue` will give you nicer report i think.

Reply:
Done.